feat(file-picker): initial design guidance (#22)
vmware-clarity/clarity-guidance: 1012-file-picker-design-guidance.md

---
category: component design guidance
state: draft
created: 2024-10-15
updated: 2024-10-21
---

# File Picker Design Guidance

> The [file picker](https://clarity.design/documentation/file-picker) component allows users to select one or more files to upload.

Reviewer: ```suggestion
The [file picker](https://clarity.design/documentation/file-picker) component enables users to select one or more files for upload, providing a straightforward interface for managing file selection.
```

## Guidance

- **Should** use compact layout in places with limited vertical space, and information density is needed; otherwise the vertical or horizontal layouts are recommended.

> - Use appropriate [file picker states](https://clarity.design/documentation/file-picker#states).

Reviewer: ```suggestion
- Design teams **should** use the appropriate [file picker states](https://clarity.design/documentation/file-picker#states) to provide clear feedback to users about their file selection and upload process, helping users make informed decisions and navigate the interaction effectively.```

> - **Should** use helper text to tell users which format the file input will accept, i.e., PNG, JPG, PDF, etc.

Reviewer: ```suggestion
- Design teams should use helper text to inform users about the accepted file formats for input, such as PNG, JPG, or PDF, ensuring users understand their options before uploading files.
```

> - **Should** follow the [forms pattern guidelines](https://clarity.design/documentation/forms).

Reviewer: ```suggestion
- Design teams **should** follow the [forms pattern guidelines](https://clarity.design/documentation/forms) to ensure consistency and clarity in the design of file picker components within the overall application.
```

---
category: component design guidance
state: draft
created: 2024-10-15
updated: 2024-10-21
---

# File Picker Design Guidance

> [File Picker](https://clarity.design/documentation/file-picker) component allow users to select one or more files to upload.

Reviewer 2: ```suggestion
The [file picker](https://clarity.design/documentation/file-picker) component allows users to select one or more files to upload.
```

## Guidance

> Commonly found in [Forms](https://clarity.design/documentation/forms) and consists of a label, a button and an optional helper message. File picker component is available in `Vertical`, `Horizontal`, and `Compact` layouts.

Reviewer 2: ```suggestion
```

This is just repeated documentation. The user can go to link above.

> - **Should** use `Compact` layout in places with limited vertical space, and information density is needed; otherwise the `Vertical` or `Horizontal` layouts are recommended.

Reviewer 2: ```suggestion
- **Should** use compact layout in places with limited vertical space, and information density is needed; otherwise the vertical or horizontal layouts are recommended.
```

- Use appropriate states for file picker. [States supported for file picker](https://clarity.design/documentation/file-picker#states).
- Recommended to use helper text to tell users which format the file input will accept, i.e., PNG, JPG, PDF, etc.

> 
> Follow usage guidelines for [Forms pattern](https://clarity.design/documentation/forms).

Reviewer 2: ```suggestion
- **Should** follow the [forms pattern guidelines](https://clarity.design/documentation/forms).
```
